docker-compose.yml

```diff
@@ -1,69 +1,69 @@
 version: '3.4'
+
 services:
+  db:
+    image: mysql:5.7
+    environment:
+      - MYSQL_DATABASE=bolt
+      - MYSQL_USER=bolt
+      - MYSQL_PASSWORD=!ChangeMe!
+      - MYSQL_ROOT_PASSWORD=!ChangeMe!
```

Comment:
I'm not sure if that makes a difference for credential exposure. If I'm interpreting https://symfony.com/doc/current/configuration.html#overriding-environment-values-via-env-local correctly .env should be checked in nowadays.

Reply:
I don't see what this changes with your solution, the `.env` file will be versioned in any case. Or otherwise I don't understand what you mean.

For me the docker-compose.yml file is really only to be used for development od the project or to run the tests in a CI.

In any case, when going into production without docker, we will simply not send the `docker` folder and the `docker-compose.yml` file to production. So changing places its variables is not necessary.

If you use docker images in production, it is not advisable to use a database image in production but a managed service.

And if you want to add the variables in the .env already in place, it is dedicated to the Symfony application and it would be a shame to add variables which are only used for development and only used for docker-compose.

And even if it is not advisable to use docker-compose in production, it is still possible. In that event best practices and to have a dedicated docker-compose.prod.yml file for production.

Reply:
I think what @djboboch is trying to say is to add to your `.env` something like:
```
DATABASE_HOST=db
DATABASE_USERNAME=bolt
DATABASE_PASSWORD=bolt
DATABASE_NAME=bolt

DATABASE_URL=mysql://${DATABASE_USERNAME}:${DATABASE_PASSWORD}@${DATABASE_HOST}:3306/${DATABASE_NAME}
```

Then in your `docker-compose.yml` you can do:
```
    environment:
      MYSQL_DATABASE: ${DATABASE_NAME}
      MYSQL_USER: ${DATABASE_USERNAME}
      MYSQL_PASSWORD: ${DATABASE_PASSWORD}
      MYSQL_RANDOM_ROOT_PASSWORD: 'yes'
```

Which then allows you to automatically set up a dev environment without setting your .env.local

Reply:
And if you want to avoid issue #36, you could manually set a env file specifically for Docker using `env_file` in docker-compose.yml

Reply:
Yes that's what I understood @JKetelaar.

For me in a Symfony project, the `.env` file should be used for the Symfony project.

```
DATABASE_USERNAME = bolt
DATABASE_PASSWORD = bolt
DATABASE_NAME = bolt
```
It's 3 variables are just used for the development environment with the `docker-compose.yml` file.
Good practice is to use docker-compose only locally and especially not on a production server.

So the `.env` file as I wrote:
- is not polluted by Docker Compose variables (local environment) when you want to use Composer.
- is not polluted by Docker Compose variables (local environment) when you want to put into production
- is easy to modify if you want to use SQLLite (so no need to remove the variables you want me to add)

Everything is not good to put everything in variable, otherwise it becomes a mess in the project and you bring bugs which should not exist.

Reply:
If you put all your environment variables in the `env` file, there may be security issues.

Varibles for Docker Compose that will be sent to the Symfony application and vice versa.

The best is to have 2 separate files or an `.env` file for the Symfony application and declare the docker environment variables in the `docker-compose.yml` file (solution I chose).

Reply:
Oh yeah, I agree, don't get me wrong, just made pointed out what I think this person meant!

Reply:
Thanks to you, at least I was able to be clearer in my words